test/unit/host_test.rb

```diff
@@ -2255,12 +2255,13 @@ def to_managed!
     assert_empty host.fact_values.reload
   end
 
-  test 'reports are deleted when build set to true' do
+  test 'reports and host.last_report are deleted when build set to true' do
     host = FactoryGirl.create(:host, :with_reports)
     assert host.reports.present?
     refute host.build?
+    refute host.last_report.nil?
```

Comment:
it looks like  FactoryGirl.create(:host, :with_reports) does not set latst_report